ethereum/statetransition/src/main/java/tech/pegasys/artemis/statetransition/util/ForkChoiceUtil.java

```diff
@@ -45,26 +43,6 @@
 
 public class ForkChoiceUtil {
 
-  public static Store get_genesis_store(BeaconStateWithCache genesis_state) {
-    BeaconBlock genesis_block = new BeaconBlock(genesis_state.hash_tree_root());
-    Bytes32 root = genesis_block.signing_root("signature");
-    Checkpoint justified_checkpoint = new Checkpoint(UnsignedLong.valueOf(GENESIS_EPOCH), root);
-    Checkpoint finalized_checkpoint = new Checkpoint(UnsignedLong.valueOf(GENESIS_EPOCH), root);
-    ConcurrentHashMap<Bytes32, BeaconBlock> blocks = new ConcurrentHashMap<>();
-    ConcurrentHashMap<Bytes32, BeaconState> block_states = new ConcurrentHashMap<>();
-    ConcurrentHashMap<Checkpoint, BeaconState> checkpoint_states = new ConcurrentHashMap<>();
-    blocks.put(root, genesis_block);
-    block_states.put(root, new BeaconStateWithCache(genesis_state));
-    checkpoint_states.put(justified_checkpoint, new BeaconStateWithCache(genesis_state));
```

Comment:
I wasn't sure why we were creating copies of the state here, so I removed the copying logic.  But if there's a good reason, I can add it back in.

Reply:
For maximum caution and to minimize the functional changes here, I've put the copying back in and added a ticket to look into better management of this copy logic: https://pegasys1.atlassian.net/browse/BC-124

Reply:
We probably only need to make one copy of the provided state rather than two, but we do need to ensure that that state added to the `Store` isn't going to wind up getting mutated after it's added.  Ideally `BeaconState` would be immutable to avoid any risk but that will take a fair bit of refactoring so we can't manage it just yet.

Not sure of the specifics of where this is called from so we *might* get away without copying here, but it seems like a good idea to be defensive since we get very hard to track down bugs if things in the store are later mutated.